src/ledger/hashmap_db.zig

```diff
@@ -174,15 +186,30 @@ pub fn SharedHashMapDB(comptime column_families: []const ColumnFamily) type {
         pub const WriteBatch = struct {
             allocator: Allocator,
             instructions: std.ArrayListUnmanaged(Instruction),
+            executed: *bool,
```

Comment:
How come this guy is a pointer to a bool instead of just a bool?

Reply:
The deinit logic of write batch depends on whether or not it has been executed. If you use the wrong logic, there may be memory errors.

The goal of using a pointer was to make it safe to call deinit from the context where it is initialized, even if it is passed by value into the context where it is executed.